domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -352,6 +395,114 @@ class ExplorationProgressController @Inject constructor(
     }
   }
 
+  /**
+   *  Checks if checkpointing is enabled, if checkpointing is enabled this function creates a
+   *  checkpoint with the latest progress and saves it using [ExplorationCheckpointController].
+   *
+   *  This function also waits for the save operation to complete, upon completion this function
+   *  uses the function [processSaveCheckpointResult] to mark the exploration as
+   *  IN_PROGRESS_SAVED or IN_PROGRESS_NOT_SAVED depending upon the result.
+   */
+  private fun saveExplorationCheckpoint() {
+    // Do not save checkpoints if checkpointing is not enabled. This is expected to happen when
+    // the current exploration has been already completed previously.
+    if (!explorationProgress.isLightweightCheckpointingEnabled) return
```

Comment:
Maybe it might be clearer to instead pass in a value that indicates whether exploration progress should be saved, such as ``shouldSavePartialProgress``. The current name reads more as "is this feature enabled" rather than providing a condition under which the feature behaves differently. Ditto everywhere (including the parameter name for startPlayingExploration).

Reply:
done